src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp

```diff
@@ -848,11 +880,15 @@ bool AnalogWrapper::close()
         RateThread::stop();
     }
 
-    //RateThread::stop();
-
     detachAll();
     removeHandlers();
 
+    if(subDeviceOwned)
+    {
+        subDeviceOwned->close();
+        delete subDeviceOwned;
```

Comment:
Not fundamental, but it is always good practice to set to `YARP_NULLPTR` a pointer after the delete.